### PR TITLE
Enabling `ruff` C416

### PR DESCRIPTION
**Pull request by jamesbraza**
# Summary

I decided to decompose my PR on incorporating `itertools.chain` a bit, and some of it can be automated by [C416](https://docs.astral.sh/ruff/rules/unnecessary-comprehension/). So this PR enables [C416](https://docs.astral.sh/ruff/rules/unnecessary-comprehension/) in `ruff` to simplify code.

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -95,7 +95,7 @@ replacement = 'src="https://raw.githubusercontent.com/encode/httpx/master/\1"'
 
 # https://beta.ruff.rs/docs/configuration/#using-rufftoml
 [tool.ruff]
-select = ["E", "F", "I", "B", "PIE"]
+select = ["E", "F", "I", "B", "PIE", "C416"]
```
Comment (tomchristie) [inline pyproject.toml:98]: I'm neither for or against these code changes, it's probably a neater style yeah.

Tho I'd prefer we stick with minimal confic where possible, so perhaps we can aim for a minimal `select` hete.

Comment (tomchristie) [inline pyproject.toml:98]:
```suggestion
select = ["E", "F", "I", "B", "PIE"]
```

**Review by tomchristie — APPROVED** [review]
Sure thing, that's neater.

**Comment by tomchristie** [comment]
I've left the config simple (the less config options the better) but included the suggested changes.
Thanks. 🙏

**Comment by jamesbraza** [comment]
Okay sounds good, thanks!

**Pull request merged**